self-hosted-runner.md

```diff
@@ -1,40 +1,51 @@
 
-For a more permanent use of the runner, `run.sh` can be used as a background process, this can be done with the help of [start-runner.sh](start-runner.sh) and [stop-runner.sh](stop-runner.sh). Put these inside the folder of the runner and then execute [start-runner.sh](start-runner.sh) to start the runner or stop the runner with [stop-runner.sh](stop-runner.sh). Something to note is on the start of the runner the console output from `run.sh` will be directed to a log file.
+**When the runner is setup it will print the** `pid` **of the self-hosted runner process.** To check if the runner is active check [here](https://github.com/diku-dk/futhark/settings/actions/runners), on this page the runners status can be seen. Another thing to note is the flag `--url` can be used to change the repository by setting it to the url of another repository.
 
-### Running the runner using Systemd
-An alternative is using `svc.sh` which is Githubs way of setting up the runner as a service with the use of `systemd`. To do this put the shell script [svc-install.sh](svc-install.sh) inside the self-hosted runner folder and execute it.
-```
-chmod +x svc-install.sh && sudo ./svc-install.sh
+### Starting the runner
+In case the self-hosted runner stops execute the command.
 ```
-This script will perform `sudo ./svc.sh install` and set up environment variables for that service. Then to start the service do.
-```
-sudo ./svc.sh start
+python3 runner.py --start
+``` 
+This command can also be used to restart the runner. When the runner has started the runner will put its log inside a log.txt file with the current date in the name.
+
+### Removal
+When removing the runner go [here](https://github.com/diku-dk/futhark/settings/actions/runners) and click the three dots on the right side of the status of the runner that will be removed. Here the command `./config.sh remove --token XXXXXXXXXXXXXXXXXXXXXXXXXXXXX`, use the token to execute the folloing command with token from before.
 ```
-To stop it do `sudo ./svc.sh stop`, to uninstall it do `sudo ./svc.sh uninstall` and to check the status of the service do `sudo ./svc.sh status`. For further reading about this look at the [Configuring the self-hosted runner application as a service](https://docs.github.com/en/actions/hosting-your-own-runners/configuring-the-self-hosted-runner-application-as-a-service?platform=linux) article.
+python3 runner.py --remove XXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+``` 
 
+## Notes about `runner.py`
+- Currently the flag `--disableupdate` is not used, so the runner will automatically update by itself.
+- The runner group used is `Default` meaning any repository from the organization it belongs to can use the runner. This can be changed using the `--runnergroup` flag.
```

Comment:
This is not great.  Can we restrict it to the single repository it is configured against?

Reply:
It should be possible to do so via organization settings -> actions -> runner groups, this should be the [link](https://github.com/organizations/diku-dk/settings/actions/runner-groups) to the page. Something to note is diku-dk has to be owned by an enterprise account to do so. I read about it [here](https://docs.github.com/en/actions/hosting-your-own-runners/managing-access-to-self-hosted-runners-using-groups).

Reply:
I do not know how I would test this since I do not have an Enterprise Account.

Reply:
Are you sure that any repository belonging to diku-dk can use these runners?  They don't show up for any of the others.

Reply:
Then I am probably wrong, this was what I could gather when reading about it, since I could not test it. I would assume since we set `--runnergroup=Default` then it would show up [here](https://i.imgur.com/GLwfx03.png) under Default? (found here [page](https://github.com/organizations/diku-dk/settings/actions/runner-groups)) But maybe when going directly to the repository to set up a runner then the runner does get added to here?

Reply:
Sounds right.

Reply:
I believe, I am confusing an [orginization level hosted runner with a repository level hosted](https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners) I can see in the "[Add self-hosted runners](https://docs.github.com/en/github-ae@latest/actions/hosting-your-own-runners/adding-self-hosted-runners#adding-a-self-hosted-runner-to-a-repository)" article it mentions the difference... Sorry for the confusion, I am sure what I wrote is wrong.